### PR TITLE
 JAMES-2470 Make LDAP a class rule in CassandraLdapJamesServerTest 

### PR DESCRIPTION
**Pull request by chibenwa**
Make LDAP docker container a class rule in CassandraLdapJamesServerTest (will be done separately: I have a failure)

```
[INFO] Apache James :: Server :: Cassandra/Ldap - guice injection SUCCESS [01:51 min]
```

### CODE REVIEW

**Comment by chibenwa** [comment]
```
[INFO] Apache James :: Server :: Cassandra/Ldap - guice injection SUCCESS [01:26 min]
```

Waiting for build results for knowing benefits...

Ps: Now with these changes, I can run LDAP tests on my computer...

File: server/testing/src/main/java/org/apache/james/util/docker/SwarmGenericContainer.java [hunk 1]
```diff
@@ -63,6 +63,10 @@ public SwarmGenericContainer(ImageFromDockerfile imageFromDockerfile) {
             logAndCheckSkipTest(e);
         }
     }
+
+    public SwarmGenericContainer(GenericContainer<?> container) {
```
[the remark this answers is not in the document]
Comment (chibenwa) [reply server/testing/src/main/java/org/apache/james/util/docker/SwarmGenericContainer.java]: Let's call it "JamesContainer" nd remove the now useless "withAffinity" stuff...

File: server/testing/src/main/java/org/apache/james/util/docker/JamesContainer.java [hunk 2]
```diff
@@ -161,8 +134,18 @@ public boolean tryConnect(int port) {
     }
 
     @Override
-    public Statement apply(Statement statement, Description description) {
-        return container.apply(statement, description);
+    public Statement apply(Statement base, Description description) {
```
Comment (mbaechler) [inline server/testing/src/main/java/org/apache/james/util/docker/JamesContainer.java]: why this change ?

File: server/testing/src/main/java/org/apache/james/util/docker/JamesContainer.java [hunk 3]
```diff
@@ -35,34 +34,33 @@
...
 import org.testcontainers.images.builder.ImageFromDockerfile;
-import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
 
-import com.github.dockerjava.api.command.InspectContainerResponse;
-import com.google.common.base.Strings;
-
-public class SwarmGenericContainer implements TestRule {
-    private static final Logger LOGGER = LoggerFactory.getLogger(SwarmGenericContainer.class);
-    private static final String DOCKER_CONTAINER = "DOCKER_CONTAINER";
+public class JamesContainer implements TestRule {
```
Comment (mbaechler) [inline server/testing/src/main/java/org/apache/james/util/docker/JamesContainer.java]: Bad Idea, how are you going to call the class the put James into a Container ? (: 
What about TestContainerRule ?

File: server/testing/src/main/java/org/apache/james/util/docker/JamesContainer.java (hunk 2 above)
Comment (chibenwa) [reply server/testing/src/main/java/org/apache/james/util/docker/JamesContainer.java]: container::apply is deprecated.

Cf: static policy of TestContainers guys.

**Comment by trantienduchn** [comment]
```
[a33c3521bbd4ecea66694f2dd8374c0f1242804c] Results :
[a33c3521bbd4ecea66694f2dd8374c0f1242804c] 
[a33c3521bbd4ecea66694f2dd8374c0f1242804c] Tests in error: 
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.containsWithGetUserShouldBeTrue:164 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.containsWithGetUserShouldBeTrueWithVirtualHosting:170 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.knownUserShouldBeAbleToLogInWhenPasswordIsCorrect:99 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.knownUserShouldBeAbleToLogInWhenPasswordIsCorrectWithVirtualHosting:123 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.knownUserShouldNotBeAbleToLogInWhenPasswordIsNotCorrect:105 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.knownUserShouldNotBeAbleToLogInWhenPasswordIsNotCorrectWithVirtualHosting:146 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.specialCharacterInUserInputShouldBeSanitized:178 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.testShouldStillWorksAfterRestartingLDAP:129 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.unknownUserShouldNotBeAbleToLogIn:111 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.unknownUserShouldNotBeAbleToLogInWhenPasswordIsCorrect:117 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.unknownUserShouldNotBeAbleToLogInWhenPasswordIsCorrectWithVirtualHosting:158 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c]   ReadOnlyUsersLDAPRepositoryTest.unknownUserShouldNotBeAbleToLogInWithVirtualHosting:152 » UsersRepository
[a33c3521bbd4ecea66694f2dd8374c0f1242804c] 
[a33c3521bbd4ecea66694f2dd8374c0f1242804c] Tests run: 12, Failures: 0, Errors: 12, Skipped: 0
[a33c3521bbd4ecea66694f2dd8374c0f1242804c] 
```

**Comment by chibenwa** [comment]
Yes, I have no clue about this failure... Works great locally.

**Comment by trantienduchn** [comment]
```
[b582464cea9bcb9b16662246280a351a3c0aef46] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 7.231 sec <<< FAILURE! - in org.apache.james.user.ldap.ReadOnlyUsersLDAPRepositoryTest
[b582464cea9bcb9b16662246280a351a3c0aef46] org.apache.james.user.ldap.ReadOnlyUsersLDAPRepositoryTest  Time elapsed: 7.231 sec  <<< ERROR!
[b582464cea9bcb9b16662246280a351a3c0aef46] org.testcontainers.containers.ContainerFetchException: Can't get Docker image: org.testcontainers.images.builder.ImageFromDockerfile@73cd37c0
[b582464cea9bcb9b16662246280a351a3c0aef46] Caused by: com.github.dockerjava.api.exception.DockerClientException: Could not build image: lstat james-project/server/data/data-ldap-integration-testing/target/test-classes/ldif-files/populate.ldif: no such file or directory
[b582464cea9bcb9b16662246280a351a3c0aef46] 
[b582464cea9bcb9b16662246280a351a3c0aef46] 
[b582464cea9bcb9b16662246280a351a3c0aef46] Results :
[b582464cea9bcb9b16662246280a351a3c0aef46] 
[b582464cea9bcb9b16662246280a351a3c0aef46] Tests in error: 
[b582464cea9bcb9b16662246280a351a3c0aef46]   ReadOnlyUsersLDAPRepositoryTest » ContainerFetch Can't get Docker image: org.t...
[b582464cea9bcb9b16662246280a351a3c0aef46] 
[b582464cea9bcb9b16662246280a351a3c0aef46] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
I really don't succeed to make this work so let's forget about it
